packages/client/eslint.config.js

```diff
@@ -0,0 +1,83 @@
+import path from "node:path"
+import { fileURLToPath } from "node:url"
+
+import { fixupConfigRules, fixupPluginRules } from "@eslint/compat"
+import { FlatCompat } from "@eslint/eslintrc"
+import js from "@eslint/js"
+import typescriptEslint from "@typescript-eslint/eslint-plugin"
+import tsParser from "@typescript-eslint/parser"
+import react from "eslint-plugin-react"
+import simpleImportSort from "eslint-plugin-simple-import-sort"
+import globals from "globals"
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: js.configs.recommended,
+  allConfig: js.configs.all,
+})
+
+const fixedUpRuleExtensions = fixupConfigRules(
+  compat.extends(
+    "eslint:recommended",
+    "plugin:react/recommended",
+    "plugin:react/jsx-runtime",
+    "plugin:@typescript-eslint/recommended",
+    "prettier",
+  ),
+)
+const fixedUpTsRules = fixupPluginRules(typescriptEslint)
+const fixedUpReactRules = fixupPluginRules(react)
+
+export default [
+  ...fixedUpRuleExtensions,
+  {
+    plugins: {
+      react: fixedUpReactRules,
+      "@typescript-eslint": fixedUpTsRules,
+      "simple-import-sort": simpleImportSort,
+    },
+
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.jest,
+        vi: "readonly", // for vitest, jest obviously doesn't cover this :)
+      },
+
+      parser: tsParser,
+      ecmaVersion: "latest",
+      sourceType: "module",
+    },
+
+    settings: {
+      react: {
+        version: "detect",
+      },
+    },
+
+    rules: {
```

Comment:
`rules` are exactly the same as before, thankfully so this part of the config "upgrade" was a straightforward lift-and-shift ..